src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -201,17 +203,28 @@ public Task Stop()
         private async Task ReadAndUpdateReminders()
         {
             if (stoppedCancellationTokenSource.IsCancellationRequested) return;
+            localRangeSequence++;
+            var cachedLocalRangeSequence = localRangeSequence;
 
             // try to retrieve reminder from all my subranges
-            myRange = ring.GetMyRange();
-            if (logger.IsVerbose2) logger.Verbose2("My range= {0}", myRange);
+            if (logger.IsVerbose2) logger.Verbose2($"My range= {myRange}, LocalRangeSequence{localRangeSequence}");
+
+            foreach (var reminder in localReminders.Where(r => !myRange.InRange(r.Key.GrainRef)).Select(r => r.Value))
+            {
+                if (logger.IsVerbose2)
+                    logger.Verbose2("Not in my range anymore, so removing. {0}", reminder);
+                // remove locally
+                reminder.StopReminder(logger);
+                localReminders.Remove(reminder.Identity);
```

Comment:
Good catch, I changed this many times and missed it. I think you're right and this will throw, I'll check and change if needed.